core/survey/taxon.js

```diff
@@ -63,19 +63,38 @@ export const isUnknownTaxon = R.pipe(getCode, R.equals(unknownCode))
 export const isEqual = ObjectUtils.isEqual
 
 //==== UPDATE
-export const assocVernacularName = (lang, vernacularName) => R.assocPath([keys.vernacularNames, lang], vernacularName)
+export const assocVernacularNames = (lang, vernacularNames) => R.assocPath([keys.vernacularNames, lang], vernacularNames)
+
+export const appendVernacularName = vernacularName => taxon => R.pipe(
+  getVernacularNamesByLang(TaxonVernacularName.getLang(vernacularName)),
+  R.append(vernacularName),
+  vernacularNames => assocVernacularNames(TaxonVernacularName.getLang(vernacularName), vernacularNames)(taxon)
+)(taxon)
+
+const _mergeVernacularNames = vernacularNamesArrayNew => vernacularNamesArrayExisting =>
+  R.reduce(
+    (accVernacularNames, index) => {
+      const vernacularNameNew = R.prop(index, vernacularNamesArrayNew)
+      const vernacularNameExisting = R.prop(index, vernacularNamesArrayExisting)
+      const vernacularNameUpdated = vernacularNameNew && vernacularNameExisting
+        ? TaxonVernacularName.mergeProps(vernacularNameNew)(vernacularNameExisting) // merge new vernacular name into existing one
+        : vernacularNameNew
+          ? vernacularNameNew // there is no existing vernacular name, take the new one
+          : vernacularNameExisting // take the existing one
+      return R.append(vernacularNameUpdated, accVernacularNames)
```

Comment:
Can we use vernacularNameNew || vernacularNameExisting here?

Reply:
right